test/src/kernels/PhiZeroKernel.C

```diff
@@ -0,0 +1,66 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PhiZeroKernel.h"
+
+registerMooseObject("MooseTestApp", PhiZeroKernel);
+
+InputParameters
+PhiZeroKernel::validParams()
+{
+  InputParameters params = NullKernel::validParams();
+  params.addClassDescription("Test for phi zero values");
+  return params;
+}
+
+PhiZeroKernel::PhiZeroKernel(const InputParameters & parameters)
+  : NullKernel(parameters), _second_phi(_assembly.secondPhi(_var))
```

Comment:
Why I can't get _second_phi.size() > 0? In my tests, I use second-order element and second-order lagrangian function

Reply:
Just calling `secondPhi` in `Assembly` is not enough to get MOOSE to do second derivatives. You need to actually request something like `_var.secondSln()` in order to get it to happen. A hacky way that may also work is to directly call `_assembly.feSecondPhi<Real>(FEType(LAGRANGE,FIRST))`

Reply:
thanks, now it works!